pkg/repositories/models/signal.go

```diff
@@ -0,0 +1,15 @@
+package models
+
+// Signal primary key
+type SignalKey struct {
+	ExecutionKey
+	SignalID string `gorm:"primary_key;index" valid:"length(0|255)"`
```

Comment:
should the execution key be part of the primary key? if so, you might need to follow the [migration models](https://github.com/flyteorg/flyteadmin/blob/master/pkg/repositories/config/migration_models.go#L10,L12) approach b/c of unique index names

Reply:
OK, so I think I understand this, but just want to clarify. With the currently definition the `signals` table looks like:
```
flyteadmin=# \d signals
                                          Table "public.signals"
      Column       |           Type           | Collation | Nullable |               Default
-------------------+--------------------------+-----------+----------+-------------------------------------
 id                | bigint                   |           | not null | nextval('signals_id_seq'::regclass)
 created_at        | timestamp with time zone |           |          |
 updated_at        | timestamp with time zone |           |          |
 deleted_at        | timestamp with time zone |           |          |
 execution_project | text                     |           | not null |
 execution_domain  | text                     |           | not null |
 execution_name    | text                     |           | not null |
 signal_id         | text                     |           | not null |
 type              | bytea                    |           | not null |
 value             | bytea                    |           |          |
Indexes:
    "signals_pkey" PRIMARY KEY, btree (execution_project, execution_domain, execution_name, signal_id)
    "idx_signals_deleted_at" btree (deleted_at)
    "idx_signals_id" btree (id)
    "idx_signals_signal_id" btree (signal_id)
```

So the ExecutionKey is part of the primary key (as it needs to be) and we have indexes on the signal primary key, deleted_at, id, and signal_id. This does not result in the index name overlap you describe because the [ExecutionKey](https://github.com/flyteorg/flyteadmin/blob/52b5e20c37bc17ef2fcbdc7d4de8b8a6c714f0dd/pkg/repositories/models/execution.go#L13-L17) currently does not specify any indexes over it's fields (ie. `Project`, `Domain`, `Name`). If it did, we would have to create a separate model to use just in the migrations that did not include those indexes. Is that correct?

Reply:
ah okay awesome, i didn't realize that the pkey already had a deterministic & unique name. as long as you don't need an index on execution key then what you have now looks great!

>  If it did, we would have to create a separate model
yep, exactly!